src/app/home/home.component.ts

```diff
@@ -14,23 +15,36 @@ import {HttpService} from '../http.service';
   styleUrls: ['./home.component.css'],
 })
 export class HomeComponent {
-  endpoint: string = '_recommend';
   recommendation$: Observable<Recommendation>;
+  liked$: Observable<boolean>;
 
   constructor(
+    public window: Window,
     public authService: AuthService,
     public httpService: HttpService
   ) {
     this.recommendation$ = authService.authenticateWithGoogle().pipe(
+      flatMap(() => authService.getIdToken()),
       flatMap((idToken: string) =>
-        httpService.get<Recommendation>(idToken, this.endpoint)
+        httpService.get<Recommendation>(idToken, '_recommend')
       ),
       map((song: Recommendation) => this.validateRecommendation(song))
     );
+    this.liked$ = of(false);
   }
 
   /** Validates/processes song recommendation. */
   validateRecommendation({album, ...rest}: Recommendation): Recommendation {
     return {album: album ? album : 'Non-Album Single', ...rest};
   }
+
+  /** Adds the recommended song to the user's liked songs. */
+  likeRecommendation(song: Recommendation) {
+    this.liked$ = this.authService.getIdToken().pipe(
```

Comment:
try not to reassign Observables.

the "liked" observable can be thought of as a combination of two things:
1. Initially, false, and then
2. The result of clicking "like"

or, restated:
`combineLatest(`Initially, false`, `The result of clicking "like"`);`

or:

`combineLatest(startWith(false), `The result of clicking "like"`);`

"The result of clicking like" is the tricky one, though.  There are a couple of ways to deal with events in Angular:
1. `(click)="runSomeCode()"` in the template
2. `fromEvent()`

Click handlers are more canonical for Angular (it's annoying to get a reference to the button from the typescript) so they are typically used.  We can combine them with a [Subject](https://rxjs-dev.firebaseapp.com/guide/subject)

```ts
// on ts class
likeClicked = new Subject<void>();

// in template
(click)="likeClicked.next(recommendation)"
```

Now we have a Subject (which is an Observable) for when the button is clicked.  So we can use that to represent "clicking like":

```ts
likeClicked.pipe(...)
```

Summarizing, now we're at:

`combineLatest(startWith(false), `The result of` likeClicked.pipe(...));`

For that, we can transform our `likeRecommendation` code into a pipe:
```ts
  const likeRecommendation: OperatorFunction<Recommendation, boolean> = pipe(
    flatMap((song: Recommendation) => {
      return this.authService.getIdToken().pipe(flatMap(idToken: string) => {
        return [song, idToken];
      }
    },
    flatMap(([song, idToken]) => {
      this.httpService.post<Like>(idToken, song, '_like')
    }),
    map((like: Like) => !!like),
  )
```

and then use it to fill in the final blank:

`combineLatest(startWith(false), likeClicked.pipe(likeRecommendation));`

So, bringing it all together, we can make `liked$` `readonly`, and put in our constructor:

`this.liked$ = combineLatest(startWith(false), likeClicked.pipe(likeRecommendation));`